[pt][fbgemm] Turn on USE_FBGEMM on Windows env

shz0116 opened the pull request:
Summary:
As Title says. FBGEMM has recently added the support for Windows.

ghstack-source-id: 97932881

Test Plan: CI

Reviewed By: jspark1105

Differential Revision: D19738268

shz0116 commented:
@peterjc123 we are currently able to build FBGEMM on Windows for PyTorch quantization support. 
(https://circleci.com/gh/pytorch/pytorch/4477133?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).
The issue is currently on the Windows test1/2, it cannot find the dll files. Do we need to change
circleci to setup environment variables or related with python ? Any suggestion ? Thanks.
cc @jianyuh

peterjc123 commented:
> The issue is currently on the Windows test1/2, it cannot find the dll files. Do we need to change
> circleci to setup environment variables or related with python ? Any suggestion ? Thanks.

Part of the log:
```
copying torch\bin\asmjit.dll -> build\lib.win-amd64-3.6\torch\bin
copying torch\bin\fbgemm.dll -> build\lib.win-amd64-3.6\torch\bin
```

From the log, it seems that `fbgemm.dll` and `asmjit.dll` are installed to `bin` instead of `lib`, which leads to the problem.




> It always import the old version
> of pytorch. Using "python setup.py install" seems not working.

Could you please try out the following command?
```bash
python setup.py install --cmake
```

peterjc123 commented on .jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat:
```diff
@@ -90,6 +90,9 @@ if NOT "%BUILD_ENVIRONMENT%"=="" (
     xcopy /s %CONDA_PARENT_DIR%\Miniconda3\Lib\site-packages\torch %TMP_DIR_WIN%\build\torch\
 )
 
+copy %TMP_DIR_WIN%\build\torch\bin\fbgemm.dll %TMP_DIR_WIN%\build\torch\lib\
```
I don't think this is a good fix. If you want to copy them, please write the logic in CMake. Otherwise, I need to fix all the build scripts.

peterjc123 replied:
Is there any specific reason you don't want to install the DLLs to `lib`?

shz0116 replied:
> I don't think this is a good fix. If you want to copy them, please write the logic in CMake. Otherwise, I need to fix all the build scripts.
This is a test to verify the problem. I have submitted a PR to change the install path for 
FBGEMM. https://github.com/pytorch/FBGEMM/pull/296.
But how to handle asmjit.dll ? Which is a third party app.

peterjc123 replied:
What about adding a install command according to https://stackoverflow.com/a/10000327?

shz0116 replied:
Add more cmake install to address this issue in pytorch/FBGEMM#296.

peterjc123 commented:
Well, `cpuinfo.dll` is not even installed.

shz0116 commented:
> Well, `cpuinfo.dll` is not even installed.

It's built statically. No dll. After the rebase, the code cause more failure on other platforms.
I am looking at the changes now.

peterjc123 commented:
The test failure is not related to this PR. You may need to rebase it on master.

peterjc123 commented:
I am attempting to fix the test failure with https://github.com/pytorch/pytorch/pull/33403. So after that is merged, it will be okay.

peterjc123 commented on .circleci/scripts/vs_install.ps1:
```diff
@@ -8,7 +8,7 @@ $VS_INSTALL_ARGS = @("--nocache","--quiet","--wait", "--add Microsoft.VisualStud
                                                      "--add Microsoft.VisualStudio.Component.VC.Redist.14.Latest",
                                                      "--add Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core",
                                                      "--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
-                                                     "--add Microsoft.VisualStudio.Component.VC.Tools.14.11",
```
This is useless now as we now use VS2019 in build/test jobs.

peterjc123 replied:
Remember to revert the changes in this file.

shz0116 commented:
> I am attempting to fix the test failure with #33403. So after that is merged, it will be okay.

Great. I assume you mean ci/circleci: pytorch_windows_vs2019_py36_cuda10.1_test2.

peterjc123 reviewed (approved):
CI passed. The changes LGTM.

jianyuh committed:
[pt][fbgemm] Turn on USE_FBGEMM on Windows env (#297)

Summary:
Pull Request resolved: https://github.com/pytorch/FBGEMM/pull/297

Pull Request resolved: https://github.com/pytorch/pytorch/pull/33250

As Title says. FBGEMM has recently added the support for Windows.

ghstack-source-id: 97932881

Test Plan: CI

Reviewed By: jspark1105

Differential Revision: D19738268

fbshipit-source-id: fca3963eba2a684bee719e6e22eb2597367b894b
